python/datafusion/context.py

```diff
@@ -685,6 +685,14 @@ def deregister_table(self, name: str) -> None:
         """Remove a table from the session."""
         self.ctx.deregister_table(name)
 
+    def register_table_provider(self, name: str, provider: Any) -> None:
```

Comment:
You could use a protocol typehint perhaps, so that you get warnings when that attribute is missing